### DISCUSSION TITLE
历史上的今天主页

### DISCUSSION THREAD

**Discussion by llsgllsg** [Meloong-Git/PCL#4990] [自定义主页]
每天凌晨1点10分自动更新内容，随机展示一条历史上的今天
图：
![image](https://github.com/user-attachments/assets/448ce342-d235-42e6-ab92-798f294c41e6)

![image](https://github.com/user-attachments/assets/2fb61c49-ea29-4b6d-8067-bfec0a907cda)
由github actions每天自动更新，自动上传国内服务器
地址https://cn-sy1.rains3.com/123456/main.xaml
测速延迟表：
![image](https://github.com/user-attachments/assets/15770a24-9897-441e-9d4e-8fb38ea87116)

**Comment by llsgllsg** [comment]
海外响应时间
![image](https://github.com/user-attachments/assets/b0969e73-fdc6-470d-9da3-5580326f350b)

**Comment by Joker2184** [comment]
嘶 有人搞过了（

我记得源码里有历史上的今天的链接.jpg

**Reply by llsgllsg** [reply]
他（她）那个没法用我才搞的，我这自动更新，访问还快，UI也还行

**Reply by Joker2184** [reply]
o~

**Comment by LTCatt** [comment]
基于其他主页修改有获得原作者同意吗？

**Reply by llsgllsg** [reply]
我不是修改他的那个，我修改了MFn233的简单主页的部分代码，经过同意了
![image](https://github.com/user-attachments/assets/805c446d-27c9-4452-b198-1379a9685b95)

**Reply by llsgllsg** [reply]
我自动更新主页是原创脚本，自动获取API，然后在保存在国内服务器上

**Reply by llsgllsg** [reply]
仓库地址https://github.com/llsgllsg/PCLls

**Reply by llsgllsg** [reply]
gitee仓库地址https://gitee.com/llsgllsg/PCLls

**Reply by Light-Beacon** [reply]
_那我的代码呢..._

**Reply by llsgllsg** [reply]
@Light-Beacon 看邮箱

**Reply by llsgllsg** [reply]
邮箱私发给你了，就是你那个lightbeacon@bugjump.net

**Comment by llsgllsg** [comment]
![image](https://github.com/user-attachments/assets/16d3852e-f3e6-45bd-bae5-21d9bde79739)
其中[upload_to_ftp.py](https://github.com/llsgllsg/PCLls/blob/main/upload_to_ftp.py)是上传到国内服务器，加快访问速度
[update_desc.py](https://github.com/llsgllsg/PCLls/blob/main/update_desc.py)是自动获取API，并且修改主页文件
[main.xaml](https://github.com/llsgllsg/PCLls/blob/main/main.xaml)是放在github上的主页文件，这个文件会被同步到国内的服务器

**Reply by llsgllsg** [reply]
然后做了每个IP每秒10次和所有请求每秒200次的限制

**Reply by llsgllsg** [reply]
@LTCatt

**Comment by LTCatt** [comment]
没有做版本号校验（

**Reply by llsgllsg** [reply]
行，我自动部署一个更新版本校验的

**Reply by llsgllsg** [reply]
主要是当时没想要做，当时是想，每个月的的流量可以访问748,982次，就没想到做这个

**Comment by Joker2184** [comment]
请配置 一个 https://cn-sy1.rains3.com/123456/main.xaml.ini 链接做版本号校验（
随便塞点东西就行 
但是可能每天晚上需要改一次.jpg 用来判断是否更新

**Reply by MoYuan-CN** [reply]
ini 可以用 action 塞提交哈希进去，如果做了自动部署的话

**Comment by llsgllsg** [comment]
@LTCatt 做好了版本号校验

**Reply by llsgllsg** [reply]
![image](https://github.com/user-attachments/assets/4b7a4e71-714c-474a-9c0d-f238f1515a41)

**Reply by llsgllsg** [reply]
搞定

**Comment by llsgllsg** [comment]
现在每天自动更新ini，比如今天是1.1明天就是1.2

**Comment by Joker2184** [comment]
好奇问一下rainyun对象存储你买多少存储空间和流量的套餐（（（

然后可以供多少人访问.jpg

**Comment by llsgllsg** [comment]
每个月10G流量，咋了
________________________________
发件人: MR.JOKER ***@***.***>
发送时间: 2024年10月26日 9:12
收件人: Hex-Dragon/PCL2 ***@***.***>
抄送: llsgllsg ***@***.***>; Author ***@***.***>
主题: Re: [Hex-Dragon/PCL2] 历史上的今天主页 (Discussion #4990)


好奇问一下rainyun对象存储你买多少存储空间和流量的套餐（（（

然后可以供多少人访问.jpg

―

**Reply by Joker2184** [reply]
> 每个月10G流量，咋了

没啥.jpg

打算买给更新日志主页和主页市场（ 买前先问问 防止套餐不够.jpg

**Reply by llsgllsg** [reply]
有个积分白嫖的，也是10G

**Comment by LTCatt** [comment]
![image](https://github.com/user-attachments/assets/a38fc267-5607-4e94-a6d0-5d10fd65cd35)
![image](https://github.com/user-attachments/assets/9958c439-098e-438e-aefd-109359d46969)

并且有的地方感觉比较重复了……
![image](https://github.com/user-attachments/assets/b1811963-7f95-4acb-bdb2-508483b504d7)

**Reply by llsgllsg** [reply]
logo加载报错是因为logo直接从那个网站上趴下来的，有些外国网站图标就有时抽风

**Reply by llsgllsg** [reply]
作者网站是我自己把我网站搞坏了，待会修好

**Reply by LTCatt** [reply]
> logo加载报错是因为logo直接从那个网站上趴下来的，有些外国网站图标就有时抽风

那换成能用的 logo 啊……

**Comment by Joker2184** [comment]
abbb

**Comment by llsgllsg** [comment]
行，周六更换，如果周中有时间就更换
________________________________
发件人: 龙腾猫跃 ***@***.***>
发送时间: 2024年11月4日 22:58
收件人: Hex-Dragon/PCL2 ***@***.***>
抄送: llsgllsg ***@***.***>; Author ***@***.***>
主题: Re: [Hex-Dragon/PCL2] 历史上的今天主页 (Discussion #4990)


logo加载报错是因为logo直接从那个网站上趴下来的，有些外国网站图标就有时抽风

那换成能用的 logo 啊……

―

**Comment by WorldHim** [comment]
> 现在每天自动更新ini，比如今天是1.1明天就是1.2
> 

可以把版本号做成日期形式？

**Comment by llsgllsg** [comment]
每天自增1写脚本更方便啊
________________________________
发件人: WorldHim ***@***.***>
发送时间: 2024年11月5日 23:14
收件人: Hex-Dragon/PCL2 ***@***.***>
抄送: llsgllsg ***@***.***>; Author ***@***.***>
主题: Re: [Hex-Dragon/PCL2] 历史上的今天主页 (Discussion #4990)


现在每天自动更新ini，比如今天是1.1明天就是1.2

可以把版本号做成日期形式？

―

**Reply by MoYuan-CN** [reply]
有脚本怎么做都方便……

**Comment by CreeperIsASpy** [comment]
我天，有点缝合怪主页的风范 (误)
上方的字体重复变换的板块是照抄雨杉楠的炽翎主页
下方的大部分都是简单主页(虽然经过许可了)
所有的原创部分全是api脚本更新

真的，感觉有点太水了。。。
无冒犯

**Reply by llsgllsg** [reply]
谢谢你的建议，我正在重构所有代码

**Reply by CreeperIsASpy** [reply]
- 似乎有没有显示完全的句子在结尾，可能影响观感，可以在py里加一个
```
if not 源字符串.endswith('。')
    源字符串 = ''.join(源字符串.split('。').pop())
```
做处理，确保结尾句为句号。

- 另外，可以给卡片下面加上`查看详细页面`的按钮，据我所知，大多数这种 api 都会有指向相关百科的链接的。

希望你的主页能做得越来越好！再接再厉！

**Reply by llsgllsg** [reply]
…………………………

**Reply by llsgllsg** [reply]
……我找到的API都没有相关连接啊啊啊啊

**Reply by llsgllsg** [reply]
其他API都要么实名，要么付费，我一个初中生实名个P啊

**Reply by llsgllsg** [reply]
我加一个用bing搜搜实践的吧

**Reply by wuliaodexiaoluo** [reply]
> ```
> if not 源字符串.endswith('。')
>     源字符串 = ''.join(源字符串.split('。').pop())
> ```


emmm，你的代码少了冒号....
字符串也不能作为变量被赋值，另外如果只有一个内容，.join 就不是必须的，按照命名约定，最好不要有无用代码
多个内容直接连一起会出现文本顺序错乱，倒不如用 + 连接

正确写法

```
if not [变量].endswith("。"):
    # str 可以避免列表内出现数字的情况，如果确信没有数字移掉也可以
    [变量2] = str([变量].split("。").pop())
```

EDIT:所以只有尾部要加上去吗？
EDIT2:没太看懂，如果有具体需求会比较好写

**Reply by LinQingYuu** [reply]
>其他API都要么实名，要么付费，我一个初中生实名个P

你可能需要: [shimoranla/Todayinhistory](https://github.com/shimoranla/Todayinhistory)

**Reply by llsgllsg** [reply]
行，我再改改

**Reply by llsgllsg** [reply]
现在弄了个使用bing搜索相关内容的

**Reply by Joker2184** [reply]
推荐塞卡片里（
而不是外面很突兀（

**Reply by llsgllsg** [reply]
找到没法整段获取的原因了，就是API返回不全

**Reply by CreeperIsASpy** [reply]
> > 其他API都要么实名，要么付费，我一个初中生实名个P
> 
> 你可能需要: [shimoranla/Todayinhistory](https://github.com/shimoranla/Todayinhistory)

@shimoranla 看看我交的PR?

**Reply by LinQingYuu** [reply]
> > > 其他API都要么实名，要么付费，我一个初中生实名个P
> > 
> > 
> > 你可能需要: [shimoranla/Todayinhistory](https://github.com/shimoranla/Todayinhistory)
> 
> @shimoranla 看看我交的PR?

? GitHub 不是有动态自动发邮件吗
这个 PR 昨天就已经合掉了（

~好好好用 2G 网是吧~

**Comment by llsgllsg** [comment]
我有看到，我打算放弃这个主页了，我在策划新项目

________________________________
发件人: 薄奚梦灵 ***@***.***>
发送时间: 2024年12月2日 20:20
收件人: Hex-Dragon/PCL2 ***@***.***>
抄送: llsgllsg ***@***.***>; Author ***@***.***>
主题: Re: [Hex-Dragon/PCL2] 历史上的今天主页 (Discussion #4990)


其他API都要么实名，要么付费，我一个初中生实名个P

你可能需要: shimoranla/Todayinhistory<https://github.com/shimoranla/Todayinhistory>

@shimoranla<https://github.com/shimoranla> 看看我交的PR?

? GitHub 不是有动态自动发邮件吗

好好好用 2G 网是吧

―

**Comment by llsgllsg** [comment]
但还是谢谢你写了这个项目，这样主页我只是不管它了，但不会停止
________________________________
发件人: 薄奚梦灵 ***@***.***>
发送时间: 2024年12月2日 20:20
收件人: Hex-Dragon/PCL2 ***@***.***>
抄送: llsgllsg ***@***.***>; Author ***@***.***>
主题: Re: [Hex-Dragon/PCL2] 历史上的今天主页 (Discussion #4990)


其他API都要么实名，要么付费，我一个初中生实名个P

你可能需要: shimoranla/Todayinhistory<https://github.com/shimoranla/Todayinhistory>

@shimoranla<https://github.com/shimoranla> 看看我交的PR?

? GitHub 不是有动态自动发邮件吗

好好好用 2G 网是吧

―

**Comment by LTCatt** [comment]
如果不继续维护了我就不加进列表咯？

**Comment by llsgllsg** [comment]
行，新项目完成进度40%了
________________________________
发件人: 龙腾猫跃 ***@***.***>
发送时间: 2024年12月5日 21:39
收件人: Hex-Dragon/PCL2 ***@***.***>
抄送: llsgllsg ***@***.***>; Author ***@***.***>
主题: Re: [Hex-Dragon/PCL2] 历史上的今天主页 (Discussion #4990)


如果不继续维护了我就不加进列表咯？

―
